### DISCUSSION TITLE
Transliteration and spelling correction

### DISCUSSION THREAD

**Discussion by n-endashi** [machinewrapped/llm-subtrans#94] [Q&A]
It must be possible to get gpt-subtrans to ask for a transliteration instead of a translation.  My use case would be taking Japanese subtitles made by Youtube or Whisper and transliterating them to romaji or hiragana without kanji.

Secondly, when I use whisper to make Latin subtitles it makes a lot of obvious spelling mistakes like using k instead of c (which is always hard in Latin) , using w instead of v, spelling hoc with a q...  How would I ask the program to simply step through my file and look for spelling mistakes and return the spell corrected version?

**Comment by machinewrapped** [comment]
Yeah, definitely possible - have a look at `instructions (improve quality).txt`, which asks GPT to improve spelling and grammar to make a more natural-sounding translation. It can make some of the same mistakes it does in translation, e.g. mixing up who is speaking a line, but in general it does a good job.

It needs to be paired with an appropriate prompt, e.g. "Correct the spelling of these subtitles[ for movie]" ... in hindsight the prompt should probably have been part of the instructions file.

Transliteration should be pretty straightforward, make a copy of the instructions file and replace the examples with something appropriate in the languages you want to use (you can ask ChatGPT to generate the examples). The drop-down in the GUI settings just looks for files in the root folder whose name starts with "instructions".

**Reply by n-endashi** [reply]
I did as you suggested and it seemed to work.  For language to translate to I put "romanized Japanese." I gave GPT-4 the default instructions file and told it modify it for transliteration and that's what it came up with.  I didn't try it with other scripts yet.

[【ドラゴンクエスト ダイの大冒険　-魂の絆-】ダイ達と一緒に冒険へ出発だ！！【ホロライブ_戌神ころね】 - Japanese (auto-generated).srt.txt](https://github.com/machinewrapped/gpt-subtrans/files/13698039/-.-._.-.Japanese.auto-generated.srt.txt)
[【ドラゴンクエスト ダイの大冒険　-魂の絆-】ダイ達と一緒に冒険へ出発だ！！【ホロライブ_戌神ころね】 - Japanese (auto-generated)-GPT.srt.txt](https://github.com/machinewrapped/gpt-subtrans/files/13698040/-.-._.-.Japanese.auto-generated.-GPT.srt.txt)

[instructions (transliteration).txt](https://github.com/machinewrapped/gpt-subtrans/files/13698031/instructions.transliteration.txt)
